packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -1974,6 +1974,30 @@ class RenderEditable extends RenderBox with RelayoutWhenSystemFontsChangeMixin,
     _setSelection(newSelection, cause);
   }
 
+  // TODO(justinmc): Take a look if you can reuse this logic in the actions/intent
+  // system anywhere.
+  // TODO(justinmc): This behavior likely should be pivot on non-Apple plattforms.
+  /// Expand the selection to the last tapped position.
+  ///
+  /// Either base or extent will be moved to the last tapped position, whichever
+  /// is closest. The selection will never shrink or pivot, only grow.
+  void expandSelection({ required SelectionChangedCause cause }) {
```

Comment:
I realized that the expand behavior from Mac has disappeared with https://github.com/flutter/flutter/pull/90684.  I filed this issue to track that https://github.com/flutter/flutter/issues/93883.

Ideally the actions code in EditableTextState could share this logic, but I guess that won't be possible until we can do tap-based actions.